src/xtgeo/grid3d/grid.py

```diff
@@ -64,7 +63,7 @@ def _handle_import(grid_constructor, gfile, fformat=None, **kwargs):
 
     """
 
-    gfile = xtgeo._XTGeoFile(gfile, mode="wb")
+    gfile = xtgeo._XTGeoFile(gfile, mode="rb")
```

Comment:
Would be a good idea to have this changing of mode in a different commit.

Reply:
Considered it as an obvious bug, and there are several lines of code that deals with `_mode` and `_memstream` dependencies in `_XTGeoFile.`